video/out/vo_gpu_next.c

```diff
@@ -1515,84 +1521,216 @@ static void wait_events(struct vo *vo, int64_t until_time_ns)
     }
 }
 
-#if PL_API_VER < 342
-static inline void xor_hash(void *hash, pl_cache_obj obj)
+static char *cache_filepath(void *ta_ctx, char *dir, const char *prefix, uint64_t key)
 {
-    *((uint64_t *) hash) ^= obj.key;
+    bstr filename = {0};
+    bstr_xappend_asprintf(ta_ctx, &filename, "%s_%016" PRIx64, prefix, key);
+    if (!filename.len)
```

Comment:
can't fail, it aborts on OOM

Reply:
I know, I still left it, even if it is only theoretical that `vsnprintf` returns 0. I have no idea if it is possible, but doesn't harm to check.

Reply:
We rely on this everywhere. Leaving the bogus check in is just confusing, also for people who read the code and might use it as inspiration.

Reply:
ok then.